contracts/pool/src/methods/repay.rs

```diff
@@ -95,26 +93,36 @@ pub fn do_repay(
             (amount, borrower_debt_to_burn, false)
         };
 
-    let lender_part = collat_coeff
-        .mul_int(borrower_debt_to_burn)
+    let treasury_coeff = debt_coeff
+        .checked_sub(collat_coeff)
         .ok_or(Error::MathOverflowError)?;
-    let treasury_part = borrower_payback_amount
-        .checked_sub(lender_part)
+    let treasury_part = treasury_coeff
+        .mul_int(borrower_payback_amount)
         .ok_or(Error::MathOverflowError)?;
+    let lender_part = borrower_payback_amount
+        .checked_sub(treasury_part)
+        .ok_or(Error::MathOverflowError)?;
+
     let debt_token_supply_after = debt_token_supply
         .checked_sub(borrower_debt_to_burn)
         .ok_or(Error::MathOverflowError)?;
+    let who_debt_after = who_debt
+        .checked_sub(borrower_debt_to_burn)
+        .ok_or(Error::MathOverflowError)?;
 
     let treasury_address = read_treasury(env);
 
     let underlying_asset = token::Client::new(env, asset);
+    let debt_token = DebtTokenClient::new(env, &reserve.debt_token_address);
+    let s_token = STokenClient::new(env, &reserve.s_token_address);
 
-    underlying_asset.transfer(who, &reserve.s_token_address, &lender_part);
-    underlying_asset.transfer(who, &treasury_address, &treasury_part);
+    underlying_asset.transfer(who, &reserve.s_token_address, &borrower_payback_amount);
+    s_token.transfer_underlying_to(&treasury_address, &treasury_part);
```

Comment:
Wouldn't it be more profitable leave a underlying_asset.transfer here insted of s_token.transfer_underlying_to ?

Reply:
Yes, it will be more efficient this way. Thanks!